lib/ansible/modules/windows/win_domain_group_membership.ps1

```diff
@@ -0,0 +1,125 @@
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+try {
+    Import-Module ActiveDirectory
+}
+catch {
+    Fail-Json $result "win_domain_group_membership requires the ActiveDirectory PS module to be installed"
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$diff_mode = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+# Module control parameters
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent","pure"
+$domain_username = Get-AnsibleParam -obj $params -name "domain_username" -type "str"
+$domain_password = Get-AnsibleParam -obj $params -name "domain_password" -type "str" -failifempty ($domain_username -ne $null)
+$domain_server = Get-AnsibleParam -obj $params -name "domain_server" -type "str"
+
+# Group Membership parameters
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$members = Get-AnsibleParam -obj $params -name "members" -type "list" -failifempty $true
+
+$extra_args = @{}
+if ($domain_username -ne $null) {
+    $domain_password = ConvertTo-SecureString $domain_password -AsPlainText -Force
+    $credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $domain_username, $domain_password
+    $extra_args.Credential = $credential
+}
+if ($domain_server -ne $null) {
+    $extra_args.Server = $domain_server
+}
+
+$result = @{
+    changed = $false
+}
+if ($state -in @("present", "pure")) {
+    $result.added = @()
+}
+if ($state -in @("absent", "pure")) {
+    $result.removed = @()
+}
+if ($diff_mode) {
+    $result.diff = @{}
+}
+
+$members_before = Get-AdGroupMember -Identity $name @extra_args
+$pure_members = @()
+
+foreach ($member in $members) {
+    $group_member = Get-ADObject -Filter "SamAccountName -eq '$member' -and (ObjectClass -eq 'user' -or ObjectClass -eq 'group' -or ObjectClass -eq 'computer' -or ObjectClass -eq 'msDS-ManagedServiceAccount')"
+    if (!$group_member) {
+        Fail-Json -obj $result "Could not find domain user, group, service account or computer named $member"
+    }
+
+    if ($state -eq "pure") {
+        $pure_members += $group_member
+    }
+
+    $user_in_group = $false
+    foreach ($current_member in $members_before) {
+        if ($current_member.sid -eq $group_member.sid) {
+            $user_in_group = $true
+            break
+        }
+    }
+
+    if ($state -in @("present", "pure") -and !$user_in_group) {
+        Add-ADGroupMember -Identity $name -Members $group_member -WhatIf:$check_mode @extra_args
+        $result.added += $group_member.SamAccountName
+        $result.changed = $true
+    } elseif ($state -eq "absent" -and $user_in_group) {
+        Remove-ADGroupMember -Identity $name -Members $group_member -WhatIf:$check_mode @extra_args -Confirm:$False
+        $result.removed += $group_member.SamAccountName
```

Comment:
With the array to List this change this would be `$result.removed.Add($group_member.sAMAccountName)`